lib/kube/proxy/forwarder.go

```diff
@@ -440,14 +440,18 @@ func (f *Forwarder) withAuthStd(handler handlerWithAuthFuncStd) http.HandlerFunc
 	}, f.formatResponseError)
 }
 
-func (f *Forwarder) withAuth(handler handlerWithAuthFunc) httprouter.Handle {
+func (f *Forwarder) withAuth(handler handlerWithAuthFunc, passthrough bool) httprouter.Handle {
```

Comment:
I'm not sure that adding a `passthrough` flag is the best approach. Makes it quite hard to reason about security invariants of the handlers. Instead, can you make another variant of withAuth wrapper that does the "passthrough" and apply it to the specific handlers, rather than propagating the flag?